libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,6 +12,144 @@
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Lazily parse the blob."""
+
+        import io
+
+        try:
+            import openai
+        except ImportError:
+            raise ImportError(
+                "openai package not found, please install it with "
+                "`pip install openai`"
+            )
+        try:
+            from pydub import AudioSegment
+        except ImportError:
+            raise ImportError(
+                "pydub package not found, please install it with " "`pip install pydub`"
+            )
+
+        if is_openai_v1():
+            # api_key optional, defaults to `os.environ['AZURE_OPENAI_API_KEY']`
+            # api_version optional, defaults to `os.environ['OPENAI_API_VERSION']`
+            # azure_endpoint/base_rule optional,
+            # defaults to `os.environ['AZURE_OPENAI_ENDPOINT']`
+            client = openai.AzureOpenAI(
+                api_key=self.api_key,
+                azure_endpoint=self.base_url,
+                api_version=self.api_version,
+            )
+        else:
+            # Set the API key if provided
+            if self.api_key:
+                openai.api_key = self.api_key
+            if self.base_url:
+                openai.base_url = self.base_url
+
+        # Audio file from disk
+        audio = AudioSegment.from_file(blob.path)
+
+        # Define the duration of each chunk in minutes
+        # Need to meet 25MB size limit for Whisper API
+        chunk_duration = 20
+        chunk_duration_ms = chunk_duration * 60 * 1000
+
+        # Split the audio into chunk_duration_ms chunks
+        for split_number, i in enumerate(range(0, len(audio), chunk_duration_ms)):
+            # Audio chunk
+            chunk = audio[i : i + chunk_duration_ms]
+            # Skip chunks that are too short to transcribe
+            if chunk.duration_seconds <= self.chunk_duration_threshold:
+                continue
+            file_obj = io.BytesIO(chunk.export(format="mp3").read())
+            if blob.source is not None:
+                file_obj.name = blob.source + f"_part_{split_number}.mp3"
+            else:
+                file_obj.name = f"part_{split_number}.mp3"
+
+            # Transcribe
+            print(f"Transcribing part {split_number + 1}!")  # noqa: T201
+            attempts = 0
+            while attempts < 3:
+                try:
+                    if is_openai_v1():
+                        transcript = client.audio.transcriptions.create(
+                            model=self.deployment_id,
+                            file=file_obj,
+                            **self._create_params,
+                        )
+                    else:
+                        transcript = openai.Audio.transcribe(
+                            self.deployment_id, file_obj
+                        )
+                    break
+                except Exception as e:
```

Comment:
We should be more specific here on which exceptions we retry on. Retrying on an Authentication error for example is unlikely to help. Or if one of the inputs is invalid and raises something like a TypeError. The only cases we want to retry are cases of service-side throttling or intermittent connectivity issues - everything else should just get raised.

Reply:
I've removed the while loop retries/attempts and added a `max_retries` attribute set to a default of 3 (this can also be set by the user). This should handle retrying for any connectivity or throttling issues without throwing an error, all other Exceptions will just get raised.